From a441b748c190a46deb486275a1c74b9b3a204d42 Mon Sep 17 00:00:00 2001
From: Isabel Paredes <isabel.paredes@quantstack.net>
Date: Mon, 12 Oct 2026 16:56:19 +0200
Subject: [PATCH 1/2] Remove dummy.c

wasm-ld: error: function signature mismatch: R_useDynamicSymbols
 >>> defined as () -> void in dummy.o
 >>> defined as (i32, i32) -> i32 in $PREFIX/lib/R/lib/libR.so
wasm-ld: error: function signature mismatch: R_registerRoutines
 >>> defined as () -> void in dummy.o
 >>> defined as (i32, i32, i32, i32, i32) -> i32 in $PREFIX/lib/R/lib/libR.so
---
 MD5         |  1 -
 src/dummy.c | 11 -----------
 2 files changed, 12 deletions(-)
 delete mode 100644 src/dummy.c

diff --git a/MD5 b/MD5
index a98e9ea..ffe883f 100644
--- a/MD5
+++ b/MD5
@@ -8,6 +8,5 @@ cf4c134ec24aeee89150f73641948bcb *R/write.R
 3896dfc7f681aa50643365f9f7fddad7 *man/writePNG.Rd
 7fc91ecfbf95133433e23f2e50b4a66d *src/Makevars
 f942ed530aef5913c936cc1c89706850 *src/Makevars.win
-8a0c6919c62fdadda4f8aa303fb63169 *src/dummy.c
 83f56798efac990963378362bbc9788d *src/read.c
 9a3371af08a7450f27bcf5f5c8d08308 *src/write.c
diff --git a/src/dummy.c b/src/dummy.c
deleted file mode 100644
index 5bddef9..0000000
--- a/src/dummy.c
+++ /dev/null
@@ -1,11 +0,0 @@
-/* dummy symbols to keep superfluous CRAN checks happy
-   (this package uses NAMESPACE C-level symbol registration
-   but the checks don't get that) */
-
-extern void R_registerRoutines(void);
-extern void R_useDynamicSymbols(void);
-
-void dummy(void) {
-    R_registerRoutines();
-    R_useDynamicSymbols();
-}
-- 
2.56.0

